packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -2012,6 +2010,30 @@ class TextSelectionGestureDetectorBuilder {
   // cursor will not move on drag update.
   bool? _dragBeganOnPreviousSelection;
 
+  /// Handler for [TextSelectionGestureDetector.onTapTrackStart].
+  ///
+  /// See also:
+  ///
+  ///  * [TextSelectionGestureDetector.onTapTrackStart], which triggers this
+  ///    callback.
+  @protected
+  void onTapTrackStart() {
+    _isShiftPressed = HardwareKeyboard.instance.logicalKeysPressed
```

Comment:
Just curious what happen if shift is released mid drag

Reply:
The shift will not be reset until the next pointer down event. `_TapStatusTrackerMixin` does not call `onTapTrackReset` on a drag but it does reset `_previousButton` and `_lastTapOffset` to `null` so on the next tap down following a drag end `_consecutiveTapCount` is reset and `keysPressedOnDown` is updated to the latest value from the `HardwareKeyboard.instance`. This updated implementation I think follows that same protocol.